zarr/tests/test_storage.py

```diff
@@ -762,7 +763,8 @@ class TestNestedDirectoryStore(TestDirectoryStore, unittest.TestCase):
     def create_store(self, normalize_keys=False):
         path = tempfile.mkdtemp()
         atexit.register(atexit_rmtree, path)
-        store = NestedDirectoryStore(path, normalize_keys=normalize_keys)
+        store = NestedDirectoryStore(path, normalize_keys=normalize_keys,
+                                     key_separator='/', auto_mkdir=True)
```

Comment:
@martindurant : when I try `key_separator='/'`, I'm seeing:
```
DEBUG:s3fs:CALL: head_object - ({},) - {'Bucket': 'idr', 'Key': 'zarr/v0.1-extra/6001240.zarr.v0.1n/0/0//zarray'}
```

Reply:
What are you expecting to see?
(or are you wondering why there is logging at all?)

Reply:
I'd expect the HEAD/GET to be for `.../.zarray` rather than `...//zarray`

Reply:
aha, OK - the "." got replaced - I suppose we have to special-case for the `".x*"` keys as opposed to leaf data files

Reply:
I opened https://github.com/martindurant/zarr/pull/1 with the special casing.

Reply:
Note: this issue still exists. I've opened https://github.com/zarr-developers/zarr-python/pull/699 against the mainline.